core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -15,52 +15,207 @@
  */
 package com.google.cloud.sql.core;
 
-import static com.google.common.truth.Truth.assertThat;
-import static org.junit.Assert.assertThrows;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-import com.google.auth.oauth2.GoogleCredentials;
-import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.sql.AuthType;
```

Comment:
The bulk of this behavior lives in SqlAdminApiFetcher and AccessTokenSupplier. Could we test those interfaces directly?

Reply:
AuthType is just part of the API. We aren't testing it here.

Reply:
Also, I removed all the mocks from `core` tests. This test class truly tests only CloudSqlInstance behavior.